templates/admin/dj/index.html.twig

```diff
@@ -1,56 +1,77 @@
 
 {% block content %}
-<main class="container">
-    <h1 class="my-5">DJs</h1>
-    {{ form_start(form) }}
-    {{ form_label(form.search) }}
-    {{ form_errors(form.search) }} 
-    <div class="input-group mb-3 w-50">
-        {{ form_widget(form.search) }}
-        <button class="btn btn-outline-dark">
-            <i class="bi bi-search"></i>
-            Rechercher
-        </button>
-    {{ form_end(form) }}
-    </div>
-    <table class="table table-hover">
-        <thead>
-            <tr>
-                <th>Prénom</th>
-                <th>Nom</th>
-                <th>Matériel</th>
-                <th>Réservations prises</th>
-                <th></th>
-                <th></th>
+	<main class="container">
+		<h1 class="my-5">DJs</h1>
+		{{ form_start(form) }}
+		{{ form_label(form.search) }}
+		{{ form_errors(form.search) }}
+		<div class="input-group mb-3 w-50">
+			{{ form_widget(form.search) }}
+			<button class="btn btn-outline-dark">
+				<i class="bi bi-search"></i>
+				Rechercher
+			</button>
+			{{ form_end(form) }}
+		</div>
+		<table class="table table-hover">
+			<thead>
+				<tr>
+					<th></th>
+					<th>Prénom</th>
+					<th>Nom</th>
+					<th>Matériel</th>
+					<th>Réservations prises</th>
+					<th></th>
+					<th></th>
 
-            </tr>
-        </thead>
-        <tbody>
-        {% for artist in artists %}
-            <tr>
-                <td class="align-middle">{{ artist.firstName }}</td>
-                <td class="align-middle">{{ artist.lastName }}</td>
-                <td class="align-middle">{{ artist.equipment }}</td>
-                <td class="align-middle">{{ artist.reservations|length }}</td>
-                <td class="align-start text-end">
-                    <a class="btn btn-outline-dark btn-sm text-nowrap px-3 my-2" href="{{ path('admin_dj_show', {'id': artist.id}) }}">
-                    <i class="bi bi-card-list pe-1"></i>
-                    Détails</a></td>
-                {% if artist.user.roles == ['ROLE_USER'] %}
-                <td class="align-middle text-end"> {{ include('admin/dj/_dj_validate_form.html.twig') }} </td>
-                {% else %}
-                <td class="align-middle text-end"> {{ include('admin/dj/_dj_deactivate_form.html.twig') }} </td>
-                {% endif %}
-            </tr>
-        {% else %}
-            <tr>
-                <td colspan="7">Aucun DJ trouvé</td>
-            </tr>
-        {% endfor %}
-        </tbody>
-    </table>
-</main>
-{% endblock %}
+				</tr>
+			</thead>
+			<tbody>
+				{% for artist in artists %}
+					<tr>
+						<td class="align-middle">
+							{% set unavailabilityTest = false %}
+							{% for unavailability in artist.unavailabilities %}
+								{% if date(unavailability.dateEnd|date("d-m-Y")) > date("now"|date('d-m-Y')) %}
+									{% if (date(unavailability.dateStart|date('d-m-Y')) <= date("now"|date('d-m-Y'))) and (date(unavailability.dateEnd|date('d-m-Y')) >= date("now"|date('d-m-Y'))) %}
```

Comment:
je ne crois pas que tu aies besoin de mixer date() et |date, le second doit suffire 
et tu peux ajouter un and pour ne faire qu'un seul if, il faut éviter de les imbriquer

Reply:
ça ne fonctionne pas sans